src/celeritas/ext/detail/GenericPlacedVolume.hh

```diff
@@ -0,0 +1,262 @@
+//----------------------------------*-C++-*----------------------------------//
+// SPDX-FileCopyrightText: 2020 CERN
+// SPDX-License-Identifier: Apache-2.0
+//---------------------------------------------------------------------------//
+/*!
+ * \file GenericPlacedVolume.hh
+ * \brief Class for a generic placed volume related to G4 to VecGeom conversion
+ *
+ * Original code from G4VecGeomNav package by John Apostolakis et.al.
+ *
+ * Original source:
+ *   https://gitlab.cern.ch/VecGeom/g4vecgeomnav/-/blob/7f5d5ec3258d2b7ffbf717e4bd37a3a07285a65f/include/GenericPlacedVolume.h
+ */
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <G4LogicalVolume.hh>
+#include <VecGeom/base/Cuda.h>
+#include <VecGeom/base/Transformation3D.h>
+#include <VecGeom/volumes/LogicalVolume.h>
+#include <VecGeom/volumes/PlacedVolume.h>
+using namespace vecgeom;
+
+namespace celeritas
+{
+
+#define VECGEOM_VECTORAPI
```

Comment:
What does this doe?

Reply:
The VECGEOM_VECTORAPI defined enables the vector interface of VecGeom's classes, e.g. PlacedVolume in this case.

Reply:
Humm .. that sounds weird to be used.  Those interface are going away but more importantly they change the class declaration and thus are likely violating the ODR rules.  Why do you need it here?

Reply:
This was kept from John's class, without much thought.  This is supposed to work with VecGeom 1.2.2, which keeps the vector interfaces still.
I thought this was required by VecGeom, but I just checked, and this is only defined in a benchmark. It can probably be removed even for VecGeom 1.2.2.  I will check this.